### PR TITLE
Social: Register new CPT for Social Posts

### PR DESCRIPTION
**Pull request by spsiddarthan**
## Proposed changes:

* Added a new class `Jetpack_Social_Note` to register the custom post type
* I have added this class to the src folder of the plugin, but if need be it can be refactored into its own package in the future. 
* I have used `dashicons-welcome-write-blog` as the menu icon for now, we can tweak this as well if need be.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pdrWKz-156-p2#comment-1932
## Testing instructions:

* Add `define( 'JETPACK_SOCIAL_NOTES_ENABLED', true );` to your `wp-config.php`. If you're on JN, install the code snippets plugin and copy `define( 'JETPACK_SOCIAL_NOTES_ENABLED', true );` into it. 
* Activate Jetpack Social on this branch
* Ensure you're able to see the new CPT Social Note in the sidebar of your wp-admin and you're able to create new Social Notes. 
* Set `JETPACK_SOCIAL_NOTES_ENABLED` to false and ensure that you no longer see the new CPT in the sidebar.
Screenshot 2024-02-02 at 5 51 09 PM

### CODE REVIEW

File: projects/plugins/social/src/class-jetpack-social-note.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	 * Register the Jetpack Social Note custom post type.
+	 */
+	public static function register() {
+		if ( ! defined( 'JETPACK_SOCIAL_NOTES_ENABLED' ) || ! constant( 'JETPACK_SOCIAL_NOTES_ENABLED' ) ) {
+			return;
+		}
+
+		$args = array(
+			'public'       => true,
+			'label'        => 'Social Note',
```
Comment (pablinos) [inline projects/plugins/social/src/class-jetpack-social-note.php]: I think we might want to add in some of the other labels for the type, and make them translated too. Some examples are [in these comments](https://developer.wordpress.org/reference/functions/register_post_type/#comment-351)

File: projects/plugins/social/src/class-jetpack-social.php [hunk 2]
```diff
@@ -104,6 +104,11 @@ function () {
 				My_Jetpack_Initializer::init();
 			}
 		);
+		add_action(
+			'init',
+			array( new Automattic\Jetpack\Social\Jetpack_Social_Note(), 'register' ),
+			1
```
Comment (pablinos) [inline projects/plugins/social/src/class-jetpack-social.php]: Is there a reason to make this high priority?

File: projects/plugins/social/src/class-jetpack-social-note.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * @package automattic/jetpack-social-plugin
+ */
+
+namespace Automattic\Jetpack\Social;
+
+/**
+ * Register the Jetpack Social Note custom post type.
+ */
+class Jetpack_Social_Note {
```
Comment (pablinos) [inline projects/plugins/social/src/class-jetpack-social-note.php]: As this is namespaced, we could just call this 'Note', but also happy to leave it more explicit.

**Review by pablinos — COMMENTED** [review]
I think this is a good start! Just a couple of comments.

File: projects/plugins/social/src/class-jetpack-social.php (hunk 2 above)
Comment (spsiddarthan) [reply projects/plugins/social/src/class-jetpack-social.php]: I have changed it, it works with the default priority as well.

File: projects/plugins/social/src/class-jetpack-social-note.php (hunk 1 above)
Comment (spsiddarthan) [reply projects/plugins/social/src/class-jetpack-social-note.php]: I have done this, Paul, let me know what you think

**Review by pablinos — DISMISSED** [review]
I've noticed that on [other types](https://github.com/Automattic/jetpack/blob/e89d87262d80f1f47c006348bd7134f594683fd9/projects/plugins/jetpack/modules/custom-post-types/testimonial.php#L16) we're using hyphens rather than underscores in the post type slug/name. I don't know if that's a requirement or not!

**Review by pablinos — APPROVED** [review]

**Pull request merged**